### DISCUSSION TITLE
useFieldArray conditional rendering when using Controller

### DISCUSSION THREAD

**Discussion by ianschmitz** [react-hook-form/react-hook-form#7189] [Q&A]
I'm wondering what is the _correct way_ to do conditional rendering within a `fields.map()` from `useFieldArray`, specifically when using `Controller` components. 

Here's my example codesandbox: https://codesandbox.io/s/dazzling-marco-hc2mk?file=/src/App.tsx 

You can see when toggling the radio from `Foo` to `Bar`, there is a conditional text input that is rendered. This was done using `watch` from within the `fields.map()`, but it didn't feel right and i wanted to see if there was a different pattern I was missing here?

The [example shown](https://codesandbox.io/s/usefieldarray-conditional-2wi6f) in the [`useFieldArray` docs](https://react-hook-form.com/api/usefieldarray) wasn't for `Controller` components, and also used CSS to selectively show the element didn't feel right when using `Controller`.

Thanks!

**Comment by Elfayer** [comment]
This is how I do it too. Sounds like the right way.
I personally also turn this option to true: (see: https://react-hook-form.com/api/useform)
<img width="818" alt="Screenshot 2021-12-01 at 2 35 45 PM" src="https://user-images.githubusercontent.com/15869149/144184289-d3c63edb-9637-491f-89a5-cac5f96df204.png">

So that I do not submit things that are not visible.
Also, you could consider using `useWatch` to trigger less re-render, see: https://react-hook-form.com/api/usewatch
> Behaves similarly to the watch API, however, this will isolate re-rendering at the component level and potentially result in better performance for your application.

Note I'm no expert either, just a doc reader and lib user.

**Reply by ianschmitz** [reply]
Thanks for the help @Elfayer. I think you've pointed me in a cleaner direction already. I could separate out the render logic used in `fields.map()` into a separate component so i can use the `useWatch` hook. Cheers!

I'm curious if any others have found better patterns or is this the canonical way to do it?
